lib/processors/versionInfoGenerator.js

```diff
@@ -14,6 +16,335 @@ function getTimestamp() {
 	return year + month + day + hours + minutes;
 }
 
+/**
+ * Manifest libraries as defined in the manifest.json file
+ *
+ * @typedef {object<string, {lazy: boolean}>} ManifestLibraries
+ *
+ * sample:
+ * <pre>
+ * {
+ * 	"sap.chart": {
+ * 		"lazy": true
+ * 	},
+ * 	"sap.f": { }
+ * }
+ * </pre>
+ */
+
+/**
+ * Extracted information from a manifest's <code>sap.app</code> and <code>sap.ui5</code> sections.
+ *
+ * @typedef {object} ManifestInfo
+ *
+ * @property {string} id The library name, e.g. "lib.x"
+ * @property {string} embeddedBy the library this component is embedded in, e.g. "lib.x"
+ * @property {string[]} embeds the embedded component names, e.g. ["lib.x.sub"]
+ * @property {ManifestLibraries} libs the dependencies, e.g. {"sap.chart":{"lazy": true}, "sap.f":{}}
+ */
+
+
+/**
+ * Processes manifest resource and extracts information.
+ *
+ * @param {module:@ui5/fs.Resource} manifestResource
+ * @returns {Promise<ManifestInfo>}
+ */
+const processManifest = async (manifestResource) => {
+	const manifestContent = await manifestResource.getString();
+	const manifestObject = JSON.parse(manifestContent);
+	const manifestInfo = {};
+
+	// sap.ui5/dependencies is used for the "manifestHints/libs"
+	if (manifestObject["sap.ui5"]) {
+		const manifestDependencies = manifestObject["sap.ui5"]["dependencies"];
+		if (manifestDependencies) {
+			const libs = {};
+			for (const [libKey, libValue] of Object.entries(manifestDependencies.libs)) {
+				libs[libKey] = {};
```

Comment:
Iteration fails for manifestDependencies.libs === undefined (or null). This was already an issue of the older implementation (using Object.keys). Maybe enrich the if with a second term.

Reply:
tackled with 8eeab50